app/helpers/sanitizer_helper.rb

```diff
@@ -0,0 +1,14 @@
+# -*- encoding : utf-8 -*-
+module SanitizerHelper
+  # Strip out script tags entirely and remove unwanted HTML markup
+  def sanitize_html(text)
+    # remove script tags and their content nodes
+    scrubber = Loofah::Scrubber.new do |node|
+      node.remove if node.name == 'script'
+    end
+    output = Loofah.fragment(text).scrub!(scrubber).to_s
+
+    # sanitize HTML (strip unwanted tags leaving content intact)
+    ActionController::Base.helpers.sanitize(output)
```

Comment:
I don't fully understand what's happening here.

Here's how I'm reading it:

* `#sanitize` doesn't strip out the contents between `<script>` tags any more…
* …so we need a custom scrubber to do that.
* We manually pre-scrub the `text` because passing the `scrubber` to `sanitize` (e.g. `sanitize(text, scrubber: scrubber)`) prevents the default scrubbers removing some other stuff that we want removed.

---

* Is that a correct assessment?
* If so, why can't we construct a scrubber that does all that we want, and pass it to `sanitize`?

I'm also in two minds about adding `#sanitize_html`.

My instinct says that its quite obscure, so `#sanitize` is going to creep in because of familiarity. Because of that, maybe we should change how `sanitize` works?

Its annoying that we can't just create our own sanitizer and [set it as default](https://github.com/rails/rails/issues/28587).

What we can do is create a custom sanitzer:

```ruby
class AlaveteliWhiteListSanitizer < Rails::Html::WhiteListSanitizer
  def sanitize(html, options = {})
    scrubber = Loofah::Scrubber.new do |node|
      node.remove if node.name == 'script'
    end

    html = Loofah.fragment(html).scrub!(scrubber).to_s

    super(html, options)
  end
end
```

…and override the [method](https://github.com/rails/rails-html-sanitizer/blob/master/lib/rails-html-sanitizer.rb#L18-L20) that returns the default sanitzer class

```ruby
Rails::Html::Sanitizer.instance_eval do
  def white_list_sanitizer
    ::AlaveteliWhiteListSanitizer
  end
end
```

This means we can use `sanitize` with our desired behaviour.

```ruby
Rails::Html::Sanitizer.white_list_sanitizer
# => AlaveteliWhiteListSanitizer

helper.sanitize('<script>alert("hi!");</script>')
# => ""
```

On the other hand, it means we're monkey patching internals.

Any thoughts?